conf/up4.bess

```diff
@@ -0,0 +1,515 @@
+# vim: syntax=py
+# -*- mode: python -*-
+# SPDX-License-Identifier: Apache-2.0
+# Copyright(c) 2019 Intel Corporation
+
+from conf.parser import *
+import conf.ports as port
+import conf.sim as sim
+
+port.setup_globals()
+
+
+# ====================================================
+#	Read json config file (START HERE)
+# ====================================================
+
+Uplink = 0x01
+Downlink = 0x02
+Access = 1
+Core = 2
+FARForwardAction = 0
+FARTunnelAction = 1
+FARDropAction = 2
+FARNotifyCPAction = 4
+interfaces = ["s1u", "sgi"]
+parser = Parser('conf/spgwu.json')
+parser.parse(interfaces)
+
+# Catch sgi & s1u MAC/IPv4 addresses
+macstr_d = None
+macstr_u = None
+s1u_ip = None
+if parser.mode == 'sim':
+    macstr_d = '00:00:00:00:00:02'
+    macstr_u = '00:00:00:00:00:01'
+    s1u_ip = {0: '11.1.1.1'}
+else:
+    macstr_d = mac_by_interface(parser.sgi_ifname)
+    macstr_u = mac_by_interface(parser.s1u_ifname)
+    s1u_ip = ips_by_interface(parser.s1u_ifname)
+
+
+inet_packets = [sim.gen_inet_packet(128, macstr_u, macstr_d, '172.16.100.1', '16.0.0.1'),
+               sim.gen_inet_packet(128, macstr_u, macstr_d, '172.12.55.99', '16.0.0.1')]
+ue_packets = [sim.gen_ue_packet(128, macstr_d, macstr_u, '11.1.1.128', '11.1.1.1', '16.0.0.1', '172.16.100.1')]
+packet_generator = {'s1u': ue_packets, 'sgi': inet_packets}
+
+
+# ====================================================
+#       Core Setup
+# ====================================================
+
+
+# Initialize workers
+cores = get_process_affinity()
+for wid in xrange(parser.workers):
+    bess.add_worker(wid=wid, core=int(cores[wid % len(cores)]))
+
+
+# ====================================================
+#       Port Setup
+# ====================================================
+
+ports = {}
+
+# Before constructing the pipeline, pause bessd
+bess.pause_all()
+
+for idx, iface in enumerate(interfaces):
+    # check if source natting for a given port is required
+    try:
+        ext_addrs = parser.interfaces[iface]["ip_masquerade"]
+    except KeyError:
+        ext_addrs = None
+    except TypeError:
+        ext_addrs = None
+
+    p = port.Port(parser.interfaces[iface]["ifname"], ext_addrs)
+    if p.name in ports:
+        continue
+
+    # initialize port with the configured driver
+    p.init_port(idx, parser.mode, parser.workers)
+
+    # setup port module with auxiliary modules
+    p.setup_port(parser.ip_frag_with_eth_mtu, parser.measure, packet_generator[iface])
+
+    # Finally add entry to ports list
+    ports[p.name] = p
+
+
+# ====================================================
+# 	Shared Pipeline (DL + UL)
```

Comment:
Look into WorkerSplit module to handle 1 vs 2 workers. Might even remove NAT patch?